pkgs/development/compilers/ghc/9.2.4.nix

```diff
@@ -61,6 +61,13 @@
 , # Whether to disable the large address space allocator
   # necessary fix for iOS: https://www.reddit.com/r/haskell/comments/4ttdz1/building_an_osxi386_to_iosarm64_cross_compiler/d5qvd67/
   disableLargeAddressSpace ? stdenv.targetPlatform.isiOS
+
+, # Whether to build an unregisterised version of GHC.
+  # GHC will normally auto-detect whether it can do a registered build, but this
+  # option will force it to do an unregistered build when set to true.
+  # See https://gitlab.haskell.org/ghc/ghc/-/wikis/building/unregisterised
+  # Registerised RV64 cross-compiler currently produces programs that segfault
+  enableUnregisterised ? !stdenv.buildPlatform.isRiscV64 && stdenv.targetPlatform.isRiscV64
```

Comment:
In my testing, I found that a registerised stage 1 cross-compiler outputs programs that segfault, meaning that both stage 1 (outputs broken programs) and stage 2 (is itself broken) are unusable. I don't see the need to check `hostPlatform`, since `targetPlatform` is `RiscV64` whether we want stage 1 or 2.

There isn't an upstream issue yet, but I've been meaning to create one. I'm currently doing a native build of 9.2.8 to verify that it isn't already fixed in 9.6.2 (the only version I've tested on native so far). I'll add a link to it in the code once I create it.

Reply:
I've since discovered that native RISC-V compilers also segfault, so I'll change this to `stdenv.hostPlatform.isRiscV64 || stdenv.targetPlatform.isRiscV64` (`hostPlatform` to ensure stage 2 GHC doesn't crash, `targetPlatform` to ensure compiled Haskell programs don't crash).

Reply:
And I'll also include a link to a GHC issue.